### PR TITLE
[produce][spaceship] Add ability to fetch and display available bundle capabilities

### PR DESCRIPTION
**Pull request by ShyamLad**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

When getting Bundle ID Capabilities using `Spaceship::Portal`, we were able to get a list of both enabled capabilities and available capabilities (capabilities that are disabled and are able to be enabled for a given bundle ID). Fastlane now uses `Spaceship::ConnectAPI` to fetch bundle ID capabilities, and the response from `Spaceship::ConnectAPI::BundleId. get_capabilities` returns a list of enabled capabilities only. Internally at Google, we have tooling that depends on the ability to find what capabilities are available to be enabled. Adding this functionality will satisfy this requirement, and also allow `Spaceship::ConnectAPI` to better serve as a replacement for `Spaceship::Portal`

### Description

The Apple developer portal fetches the available capabilities for the "Edit your App ID Configuration" page using the following API: `POST https://developer.apple.com/services-account/v1/capabilities`

A request to this endpoint was added to `Spaceship::ConnectAPI::Provisioning::API` and a new `Spaceship::ConnectAPI::Model` was made called `Spaceship::ConnectAPI::Capabilities` to parse and store the response. A call to this model was made available in `Produce::Service` which was then fed back to `Produce::Service::CommandsGenerator` and stylized into a printable table.

### Testing Steps

Test can be run using:

`bundle exec rspec spaceship/spec/connect_api/models/capabilities_spec.rb`

To try out, you can run a command like this:

`bundle exec fastlane produce available_services -a com.google.foobar`

**Example CLI Printout**
```
+---------------------------------------+----------------------------------+------------------------------------------------------------+
|                                                          Available Services                                                           |
+---------------------------------------+----------------------------------+------------------------------------------------------------+
| Name                                  | ID                               | Description                                                |
+---------------------------------------+----------------------------------+------------------------------------------------------------+
| Access WiFi Information               | ACCESS_WIFI_INFORMATION          | Access Wifi Information allows an app to obtain            |
|                                       |                                  | information about the currently connected WiFi network.    |
+---------------------------------------+----------------------------------+------------------------------------------------------------+
| App Attest                            | APP_ATTEST                       | App Attest allows you to verify that an iOS app            |
|                                       |                                  | connecting to your server is legitimate, by signing the    |
|                                       |                                  | app identity upon connection.                              |
+---------------------------------------+----------------------------------+------------------------------------------------------------+
| App Groups                            | APP_GROUPS                       | App groups gives multiple apps access to shared            |
|                                       |                                  | containers and allows certain additional communication     |
|                                       |                                  | between the apps.                                          |
+---------------------------------------+----------------------------------+------------------------------------------------------------+
| Associated Domains                    | ASSOCIATED_DOMAINS               | Associated domains allow an app to be associated with a    |
|                                       |                                  | domain for specific services such as accessing Safari      |
|                                       |                                  | saved passwords and activity continuation.                 |
+---------------------------------------+----------------------------------+------------------------------------------------------------+
| AutoFill Credential Provider          | AUTOFILL_CREDENTIAL_PROVIDER     | The AutoFill credential provider allows an app to provide  |
|                                       |                                  | user names and passwords for AutoFill in Safari and other  |
|                                       |                                  | apps, with user permission.                                |
+---------------------------------------+----------------------------------+------------------------------------------------------------+
| CarPlay Audio App (CarPlay framework) | CARPLAY_PLAYABLE_CONTENT         | Enable your app to play audio content in CarPlay.          |
+---------------------------------------+----------------------------------+------------------------------------------------------------+
```

Thanks for your time! 😊

### CODE REVIEW

File: produce/lib/produce/service.rb
Comment (joshdholtz) [inline produce/lib/produce/service.rb:57]:
```suggestion
      return Spaceship::ConnectAPI::Capabilities.all
```

**Review by joshdholtz — CHANGES_REQUESTED** [review]
This looks great! Just one quick suggestion about removing some console output. After that this can be merged 💪

**Comment by ShyamLad** [comment]
@googlebot I fixed it.

**Comment by ShyamLad** [comment]
@googlebot I fixed it.

**Comment by ShyamLad** [comment]
@joshdholtz i accidentally modified the code in the comment you made using github's UI, and it made you a co-author. The google bot thinks you (me@joshholtz.com) do not have the CLA agreement. I tried reverting the commit and making the change myself but it still list you as an author 🤦  Do you mind signing the CLA?

**Comment by joshdholtz** [comment]
I've signed the CLA and I think the Google Bot is just mega confused today for some reason 😛

Don't worry about it though! I can still click the merge button 😇

**Comment by joshdholtz** [comment]
@googlebot I fixed it.

**Comment by joshdholtz** [comment]
Orrrrrrrr... my CLA was gone for some reason 😅 I don't understand computers.

**Comment by ShyamLad** [comment]
haha yeah computers are weird 😝  Ty for reviewing!

**Review by joshdholtz — APPROVED** [review]
Thanks so much for adding this! This is a great addition. Really appreciate the contribution ❤️

**Pull request merged**
